### PR TITLE
First draft of documentation

### PR DESCRIPTION
**Pull request by stojg**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,4 +2,56 @@
...
 
+Sensible proxy acts as a HTTP/HTTPS proxy that steers all incoming traffic
+to the www sub domain of the requested domain.
 
+## Theory of operation
+
+When using cloud providers it is very common that they will give you a DNS CNAME
+to a load balancer for high availability. Apex domains on the other hand should
+always point to a static IP address. Multiple DNS providers allows you to setup
+an "ALIAS" record that by passes this restriction, but for those without this
```
Comment (halkyon) [inline README.md]: "bypasses"

Comment (halkyon) [inline README.md]: Suggestion: "works around this restriction"

File: README.md [hunk 2]
```diff
@@ -2,4 +2,56 @@
...
+domain and tries to open up a connection to that domain prepended with `www.`.
+If the connection is successful, it will copy TCP traffic between the client and
+the backend acting as a middle man.
+
+For HTTPS traffic, sensible proxy relies on the Server Name Indication (SNI) TLS
+extension. This in practice means that it does not need to configured with
+SSL certificate and private key for decrypting the request.
+
+All modern and popular browsers are providing the hostname in the SNI header.
+
```
Comment (halkyon) [inline README.md:24]: Don't need 3 new lines here ;-)

File: README.md (hunk 1 above)
Comment (andrewandante) [inline README.md]: allows -> allow, by passes ->bypasses

File: README.md [hunk 3]
```diff
@@ -2,4 +2,56 @@
...
+## Theory of operation
+
+When using cloud providers it is very common that they will give you a DNS CNAME
+to a load balancer for high availability. Apex domains on the other hand should
+always point to a static IP address. Multiple DNS providers allows you to setup
+an "ALIAS" record that by passes this restriction, but for those without this
+functionality, sensible proxy was created.
+
+Sensible proxy looks at the incoming HTTP(S) connection, finds the requested
+domain and tries to open up a connection to that domain prepended with `www.`.
```
Comment (halkyon) [inline README.md]: Suggestion: "negotiates a connection to the web server listening on the www domain"

File: README.md [hunk 4]
```diff
@@ -2,4 +2,56 @@
...
+If the connection is successful, it will copy TCP traffic between the client and
+the backend acting as a middle man.
```
Comment (andrewandante) [inline README.md]: backend acting -> backend, acting

Comment (halkyon) [inline README.md]: Suggestion: "it will act as the middle man, by copying traffic between the client and the web server"

File: README.md [hunk 5]
```diff
@@ -2,4 +2,56 @@
...
+
+For HTTPS traffic, sensible proxy relies on the Server Name Indication (SNI) TLS
+extension. This in practice means that it does not need to configured with
```
Comment (andrewandante) [inline README.md]: need to configured -> need to be configured

File: README.md [hunk 6]
```diff
@@ -2,4 +2,56 @@
...
+SSL certificate and private key for decrypting the request.
```
Comment (andrewandante) [inline README.md]: with SSL certificate -> with an SSL certificate

File: README.md (hunk 5 above)
Comment (halkyon) [inline README.md]: Suggestion: "This means you don't need an SSL certificate configured on the proxy itself for it to work"

File: README.md [hunk 7]
```diff
@@ -2,4 +2,56 @@
...
+
+All modern and popular browsers are providing the hostname in the SNI header.
```
Comment (andrewandante) [inline README.md]: are providing -> provide

File: README.md [hunk 8]
```diff
@@ -2,4 +2,58 @@
...
+
+## Configuration
+
+Sensible proxy can be started without any configuration, but can be customised
+via environment variables.
+
+`HTTP_PORT` default: 80
+
+Listen port on to which relieve HTTP traffic
```
Comment (andrewandante) [inline README.md]: Actually, I think this should be:

`Listening port to receive HTTP traffic`, same below for HTTPS

**Pull request merged**